CI, DOC, TST: failure related to scipy/stats/_distn_infrastructure.py docstring

Looks like this snuck in with gh-20222 a few hours ago. It is visible in the CI there, where that docstring is touched, and also elsewhere now like in gh-20215. Maybe something tricky with the LaTeX-style math added.

```
 _________________ ERROR at setup of test_warning_calls_filters _________________
[gw0] linux -- Python 3.11.8 /opt/hostedtoolcache/Python/3.11.8/x64/bin/python
scipy/_lib/tests/test_warnings.py:74: in warning_calls
    tree = ast.parse(file.read(), filename=str(path))
        bad_filters = ['optimize/_optimize.py:1138', 'optimize/_constraints.py:393']
        bad_stacklevels = []
        base       = PosixPath('/home/runner/work/scipy/scipy/build-install/lib/python3.11/site-packages/scipy')
        file       = <_io.TextIOWrapper name='/home/runner/work/scipy/scipy/build-install/lib/python3.11/site-packages/scipy/stats/_distn_infrastructure.py' mode='r' encoding='utf-8'>
        finder     = <scipy._lib.tests.test_warnings.FindFuncs object at 0x7faa391e8cd0>
        path       = PosixPath('/home/runner/work/scipy/scipy/build-install/lib/python3.11/site-packages/scipy/stats/_distn_infrastructure.py')
        tree       = <ast.Module object at 0x7faa4f02f160>
/opt/hostedtoolcache/Python/3.11.8/x64/lib/python3.11/ast.py:50: in parse
    return compile(source, filename, mode, flags,
E     File "/home/runner/work/scipy/scipy/build-install/lib/python3.11/site-packages/scipy/stats/_distn_infrastructure.py", line 2462
E       """
E       ^^^
E   SyntaxError: invalid escape sequence '\p'
        feature_version = -1
        filename   = '/home/runner/work/scipy/scipy/build-install/lib/python3.11/site-packages/scipy/stats/_distn_infrastructure.py'
        flags      = 1024
        mode       = 'exec'
        source     = '#\n# Author:  Travis Oliphant  2002-2011 with contributions from\n#          SciPy Developers 2004-2011\n#\nfrom scip... if isinstance(value, rv_base_class):\n            distn_names.append(name)\n    return distn_names, distn_gen_names\n'
        type_comments = False
```

--- a/scipy/stats/_distn_infrastructure.py
+++ b/scipy/stats/_distn_infrastructure.py
@@ -2459,7 +2459,7 @@ class rv_continuous(rv_generic):
                  np.maximum(np.abs(data_moments), 1e-8))**2).sum()
 
     def fit(self, data, *args, **kwds):
-        """
+        r"""
         Return estimates of shape (if applicable), location, and scale
         parameters from data. The default estimation method is Maximum
         Likelihood Estimation (MLE), but Method of Moments (MM)
@@ -2559,18 +2559,18 @@ class rv_continuous(rv_generic):
 
         .. math::
 
-            l(\pmb{\\theta}; k) & = \sum
-                                    \log(f(k_u; \pmb{\\theta}))
+            l(\pmb{\theta}; k) & = \sum
+                                    \log(f(k_u; \pmb{\theta}))
                                 + \sum
-                                    \log(F(k_l; \pmb{\\theta})) \\\\
+                                    \log(F(k_l; \pmb{\theta})) \\
                                 & + \sum 
-                                    \log(1 - F(k_r; \pmb{\\theta})) \\\\
+                                    \log(1 - F(k_r; \pmb{\theta})) \\
                                 & + \sum
-                                    \log(F(k_{\\text{high}, i}; \pmb{\\theta})
-                                    - F(k_{\\text{low}, i}; \pmb{\\theta}))
+                                    \log(F(k_{\text{high}, i}; \pmb{\theta})
+                                    - F(k_{\text{low}, i}; \pmb{\theta}))
 
         where :math:`f` and :math:`F` are the pdf and cdf, respectively, of the
-        function being fitted, :math:`\pmb{\\theta}` is the parameter vector,
+        function being fitted, :math:`\pmb{\theta}` is the parameter vector,
         :math:`u` are the indices of uncensored observations,
         :math:`l` are the indices of left-censored observations,
         :math:`r` are the indices of right-censored observations,

